currency/manager_test.go

```diff
@@ -463,72 +462,54 @@ func TestEnablePair(t *testing.T) {
 	}
 }
 
-func TestIsAssetEnabled_SetAssetEnabled(t *testing.T) {
+func TestAssetEnabled(t *testing.T) {
 	t.Parallel()
 	p := initTest(t)
 
 	err := p.IsAssetEnabled(asset.Empty)
-	if !errors.Is(err, asset.ErrNotSupported) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, asset.ErrNotSupported)
-	}
+	assert.ErrorIs(t, err, asset.ErrNotSupported)
+
 	p.Pairs = nil
+
 	// Test enabling a pair when the pair manager is not initialised
 	err = p.IsAssetEnabled(asset.Spot)
-	if err == nil {
-		t.Error("unexpected result")
-	}
+	assert.ErrorIs(t, err, ErrPairManagerNotInitialised)
 
 	err = p.SetAssetEnabled(asset.Spot, true)
-	if err == nil {
-		t.Fatal("unexpected result")
-	}
+	assert.ErrorIs(t, err, ErrPairManagerNotInitialised)
 
 	// Test asset type which doesn't exist
 	p = initTest(t)
 
-	p.Pairs[asset.Spot].AssetEnabled = nil
+	p.Pairs[asset.Spot].AssetEnabled = false
 
 	err = p.IsAssetEnabled(asset.Spot)
-	if err == nil {
-		t.Error("unexpected result")
-	}
+	assert.ErrorIs(t, err, asset.ErrNotEnabled)
 
 	err = p.SetAssetEnabled(asset.Spot, false)
-	if err != nil {
-		t.Fatal(err)
-	}
+	assert.NoError(t, err)
 
 	err = p.SetAssetEnabled(asset.Spot, false)
-	if err == nil {
-		t.Fatal("unexpected result")
-	}
+	assert.NoError(t, err, "Setting to disabled twice should not error")
```

Comment:
Suggestion: I usually return an error on no-op because I find it easier to track my own whoopsies, I also like the fact if I change it by CLI it then returns that it failed. I do see the idempotency aspect of the new behaviour though. Keep this unresolved for other peoples input if you want to pass on this.

Reply:
I believe if something wants to error on it already being disabled/enabled, that's *it's* problem.
The responsibility of this function is to do what it's told 😃 
So I'm happy to add a check in api server or CLI for you to see if it's already enabled/disabled and error *there*.
Would you like that?

Reply:
No thanks, if it's going to stay like this, having that check would need to go to all potential API calls cli, scripting etc. State change responsibility is not a major concern 🤷 just easier here but it's such a minor thing.